pytensor/link/numba/dispatch/compile_ops.py

```diff
@@ -109,22 +109,28 @@ def numba_funcify_IfElse(op, **kwargs):
         @numba_basic.numba_njit
         def ifelse(cond, *args):
             if cond:
-                res = args[:n_outs]
+                selected = args[:n_outs]
             else:
-                res = args[n_outs:]
+                selected = args[n_outs:]
 
-            return res
+            # Return a tuple of copies
+            out = [None] * n_outs
+            for i in range(n_outs):
+                out[i] = selected[i].copy()
+
+            return tuple(out)
 
     else:
 
         @numba_basic.numba_njit
         def ifelse(cond, *args):
             if cond:
-                res = args[:n_outs]
+                arr = args[0]
```

Comment:
This case can be simplified to have signature `ifelse(cond, if_true, if_false)`, without need for indexing internally. 

It's unrelated to the copy change